packages/integration-sdk-cli/src/commands/generate-integration-data-sources.test.ts

```diff
@@ -0,0 +1,182 @@
+        'Dependabot vulnerability alert ingestion and Code scanning alerts',
+      defaultsToDisabled: true,
+    },
+    [INGESTION_SOURCE_IDS.FETCH_REPOS]: {
+      title: 'Fetch Alerts',
+      description: 'This is an ingestion source created fo test purposes',
+      defaultsToDisabled: false,
+    },
+  };
+
+  it('should return the ingestionConfig with empty childIngestionSources', () => {
+    const integrationSteps: IntegrationStep<IntegrationInstanceConfig>[] = [
+      {
+        id: 'fetch-vulnerability-alerts',
+        name: 'Fetch Vulnerability Alerts',
+        entities: [
+          {
+            resourceName: 'GitHub Vulnerability Alerts',
+            _type: 'github_finding',
+            _class: ['Finding'],
+          },
+        ],
+        relationships: [],
+        dependsOn: ['fetch-repos'],
+        ingestionSourceId: INGESTION_SOURCE_IDS.FINDING_ALERTS,
+        executionHandler: jest.fn(),
+      },
+    ];
+    const ingestionDataSources = generateIntegrationIngestionDataSources(
+      ingestionConfig,
+      integrationSteps,
+    );
+    // Original object doesn't change
+    expect(
+      ingestionDataSources[INGESTION_SOURCE_IDS.FINDING_ALERTS],
+    ).toMatchObject(ingestionConfig[INGESTION_SOURCE_IDS.FINDING_ALERTS]);
+    // childIngestionSources is empty because there are no steps that depends on fetch-vulnerability-alerts
+    expect(
+      ingestionDataSources[INGESTION_SOURCE_IDS.FINDING_ALERTS]
+        .childIngestionSources,
+    ).toBeEmpty();
+  });
+
+  it('should return the ingestionConfig with childIngestionSources', () => {
+    const integrationSteps: IntegrationStep<IntegrationInstanceConfig>[] = [
+      {
+        id: 'fetch-repos',
+        name: 'Fetch Repos',
+        entities: [
+          {
+            resourceName: 'Github Repo',
+            _type: 'github_repo',
+            _class: ['CodeRepo'],
+          },
+        ],
+        relationships: [],
+        dependsOn: ['fetch-account'],
+        ingestionSourceId: INGESTION_SOURCE_IDS.FETCH_REPOS,
+        executionHandler: jest.fn(),
+      },
+      {
+        id: 'fetch-vulnerability-alerts',
+        name: 'Fetch Vulnerability Alerts',
+        entities: [
+          {
+            resourceName: 'GitHub Vulnerability Alerts',
+            _type: 'github_finding',
+            _class: ['Finding'],
+          },
+        ],
+        relationships: [],
+        dependsOn: ['fetch-repos'],
+        ingestionSourceId: INGESTION_SOURCE_IDS.FINDING_ALERTS,
+        executionHandler: jest.fn(),
+      },
+      {
+        id: 'fetch-issues',
+        name: 'Fetch Issues',
+        entities: [
+          {
+            resourceName: 'GitHub Issue',
+            _type: 'github_issue',
+            _class: ['Issue'],
+          },
+        ],
+        relationships: [],
+        dependsOn: ['fetch-repos', 'fetch-users', 'fetch-collaborators'],
+        executionHandler: jest.fn(),
+      },
+      {
+        id: 'fetch-teams',
```

Comment:
Since `fetch-issues` and `fetch-teams` do not have `ingestionSourceId` they will not show up in the ingestionConfig? Is that correct?
I wonder if we want them to show up in the ingestionConfig so that they show up in the UI but as being "required". Looking at Michelle's mockups, that is the case.